crates/stc_ts_file_analyzer/src/analyzer/decl_merging.rs

```diff
@@ -97,24 +97,37 @@ impl Analyzer<'_, '_> {
             }
 
             (Type::Interface(a), Type::Interface(bi)) => {
-                // TODO: Handle the number of type parameters.
                 let mut type_params = FxHashMap::default();
-                if let Some(b_tps) = &bi.type_params {
-                    if let Some(a_tp) = &a.type_params {
+
+                match (&a.type_params, &bi.type_params) {
+                    (Some(a_tps), Some(b_tps)) => {
+                        if a_tps.params.len() != b_tps.params.len() {
+                            self.storage
+                                .report(ErrorKind::InterfaceNonIdenticalTypeParams { span: a.span() }.into());
+                            self.storage
+                                .report(ErrorKind::InterfaceNonIdenticalTypeParams { span: b.span() }.into());
+                        }
+
                         for (idx, b_tp) in b_tps.params.iter().enumerate() {
                             type_params.insert(
                                 b_tp.name.clone(),
                                 Type::Param(TypeParam {
-                                    span: a_tp.span,
-                                    name: a_tp.params[idx].name.clone(),
+                                    span: a_tps.span,
+                                    name: a_tps.params.get(idx).unwrap_or(b_tp).name.clone(),
```

Comment:
Does it matter here that `a_tps.span` can be passed when `a_tps.params.get(idx) = None` ?

Reply:
Hmm, not sure, but I don't think it matters.